storybookjs/storybook
# Add Doc Blocks overview & API reference

## What I did

- Add Doc Blocks overview & API reference pages for each block
- Remove Doc Blocks guides
- Update `docs/api/argtypes` to include information that was in `docs/writing-docs/doc-block-argstable`
- Update TOC
    - Change `Write docs > Doc blocks` from a menu to a page
    - Add `API > @storybook/blocks menu`

<table>
<tr>
	<th>Before
	<th>After
<tr>
	<td><img src="https://user-images.githubusercontent.com/486540/219272987-8bf85296-82cf-4191-b7ee-0531f39ef1a8.png" alt="Screenshot of current TOC" />
	<td><img src="https://user-images.githubusercontent.com/486540/219273086-0477866b-74c1-4ed9-b975-e37f145c2a01.png" alt="Screenshot of proposed TOC" />
</table>

### Example page

![Screenshot of ArgTypes doc block API reference](https://user-images.githubusercontent.com/486540/220531432-15010e74-e58f-4cfc-b071-4c49478bc11a.png)

### WIP

- Add/update screenshots for all doc blocks

## How to test

1. In `frontpage`, check out the `docs-content-style-tweaks` branch ([see related PR](https://github.com/storybookjs/frontpage/pull/487))
    - The content won't look right otherwise
2. Extract the docs for this branch: `yarn extract-monorepo-docs doc-blocks-docs`
3. Start the site: `GATSBY_SKIP_ADDON_PAGES=true yarn start`
4. Navigate to the updated/new pages:
    - [Doc blocks overview](http://localhost:8000/docs/react/writing-docs/doc-blocks)
    - [ArgTypes API reference](http://localhost:8000/docs/react/api/argtypes/)
    - And so on...

### What to check for:

- Correctness
- Consistent, helpful structure
- Understandability of text content
- _**Not**_ design ([separate `frontpage` PR for that](https://github.com/storybookjs/frontpage/pull/487))

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
